nav2_bt_navigator/behavior_trees/navigate_to_pose_w_replanning_monitoring_and_recovery.xml

```diff
@@ -0,0 +1,46 @@
+
+<!--
+  This Behavior Tree replans the global path periodically at 1 Hz and it also has
+  recovery actions specific to planning / control as well as general system issues.
+  This will be continuous if a kinematically valid planner is selected.
+-->
+<root main_tree_to_execute="MainTree">
+  <BehaviorTree ID="MainTree">
+    <RecoveryNode number_of_retries="6" name="NavigateRecovery">
+      <PipelineSequence name="NavigateWithReplanning">
+        <RateController hz="1.0">
+          <RecoveryNode number_of_retries="1" name="ComputePathToPose">
+            <ComputePathToPose goal="{goal}" path="{path}" planner_id="GridBased"/>
+            <ClearEntireCostmap name="ClearGlobalCostmap-Context" service_name="global_costmap/clear_entirely_global_costmap"/>
+          </RecoveryNode>
+        </RateController>
+        <ReactiveSequence name="MonitorAndFollowPath">
+          <IsPathLongerOnApproach new_path="{path}" prox_leng="0.0" length_factor="1.0">
+            <RetryUntilSuccessful num_attempts="1">
+              <SequenceStar name="CancelingControlAndWait">
+                <CancelControl name="ControlCancel"/>
+                <Wait wait_duration="5"/>
+              </SequenceStar>
+            </RetryUntilSuccessful>
```

Comment:
Please base this BT off of the most recent ones, there are excessive `GoalUpdated` checks that don't do anything. Pull in `main`

Reply:
Sure

Reply:
`main` also has the `GoalUpdated` checks. I added the BT based on the recent changes in the `main`.

Reply:
maybe this point I didn't get you. I did use the recent changes for the BT.